### ISSUE TITLE
Cucumber Data Tables are not displayed in Allure report

### ISSUE THREAD

**Issue by mareru** [webdriverio/webdriverio#6895]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 7.6.0
 - **Mode:** [WDIO Testrunner]
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 14.16.1
 - **NPM version:** 6.14.8
 - **Browser name and version:** Chrome 90
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):**
```
  "devDependencies": {
    "@cucumber/cucumber": "^7.2.1",
    "@percy/cli": "^1.0.0-beta.50",
    "@percy/webdriverio": "^2.0.0",
    "@types/chai": "^4.2.18",
    "@types/node": "^15.6.0",
    "@typescript-eslint/eslint-plugin": "^4.24.0",
    "@typescript-eslint/parser": "^4.24.0",
    "allure-commandline": "^2.13.8",
    "app-root-path": "^3.0.0",
    "chai": "^4.3.4",
    "eslint": "^7.26.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^3.4.0",
    "husky": "^6.0.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.3.0",
    "ts-node": "^9.1.1",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.2.4",
    "webdriverio": "^7.6.0"
  },
  "dependencies": {
    "@wdio/allure-reporter": "^7.6.0",
    "@wdio/cli": "^7.6.0",
    "@wdio/cucumber-framework": "^7.6.0",
    "@wdio/local-runner": "^7.6.0",
    "@wdio/selenium-standalone-service": "^7.6.0",
    "@wdio/spec-reporter": "^7.6.0",
    "@wdio/sync": "^7.6.0",
    "@wdio/types": "^7.6.0"
  }
```

**Config of WebdriverIO**
```
  reporters: [
    'spec',
    [
      'allure',
      {
        outputDir: 'allure-results',
        disableWebdriverStepsReporting: true,
        disableWebdriverScreenshotsReporting: false,
        useCucumberStepReporter: true,
      },
    ],
  ],
```

**Describe the bug**
When I use Cucumber Data Table, then in the allure report I am missing the information from data table.

**To Reproduce**

1. Having this kind of feature file:

```
  Scenario: Register new user without accepting terms and conditions
    And I fill out registration form with:
      | title              | miss         |
      | firstName          | Jane         |
      | lastName           | Doe          |
      | email              | test@test.com |
      | password           | 123456       |
      | confirmPassword    | 123456       |
      | consent            | check        |
      | termsAndConditions | do not check |
    Then Register button is disabled
```

2.  run the test
3.  generate allure report `allure generate ./allure-results --clean && allure open`

**Expected behavior**
I expect the data from data table to be visible in the allure report.
Instead I have this in the report:

![image](https://user-images.githubusercontent.com/6464981/119208745-cdcafd00-baa3-11eb-87c7-95d76287d5f4.png)

**Additional context**
Spec reporter behaves as expected. 

![image](https://user-images.githubusercontent.com/6464981/119209273-05d33f80-baa6-11eb-9216-9044858550e3.png)

**Comment by christian-bromann** [comment]
@BorisOsipov do you know if Allure supports tables?

**Comment by BorisOsipov** [comment]
@christian-bromann I'm not sure. But it can be done as step attachment.

**Comment by mareru** [comment]
Please also note that Data tables can be used in combination with `Scenario Outlines`.  
So I could have something like this:  
```
  Scenario Outline: Register new user with various invalid emails (<invalidEmail>)
    And I fill out registration form with:
      | title              | miss           |
      | firstName          | Jane           |
      | lastName           | Doe            |
      | email              | <invalidEmail> |
      | password           | 123456         |
      | confirmPassword    | 123456         |
      | consent            | check          |
      | termsAndConditions | check          |
    Then I can see "Please correct the errors below." error message on the top of the page
    And I can see registration form error messages
      | email | Please enter a valid email |
    Examples:
      | invalidEmail   |
      | ecx 1@test.com |
      | ecx@test       |
      | ecx@           |
```
In this case I would expect to see in the Allure report the same as in spec:  
```
[chrome 90.0.4430.212 windows #0-0]     Register new user with various invalid emails (ecx 1@test.com)
[chrome 90.0.4430.212 windows #0-0]        ✓ Given I visit home page
[chrome 90.0.4430.212 windows #0-0]        ✓ When I navigate to registration page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I fill out registration form with:
[chrome 90.0.4430.212 windows #0-0]          │ title              │ miss           │
[chrome 90.0.4430.212 windows #0-0]          │ firstName          │ Jane           │
[chrome 90.0.4430.212 windows #0-0]          │ lastName           │ Doe            │
[chrome 90.0.4430.212 windows #0-0]          │ email              │ ecx 1@test.com │
[chrome 90.0.4430.212 windows #0-0]          │ password           │ 123456         │
[chrome 90.0.4430.212 windows #0-0]          │ confirmPassword    │ 123456         │
[chrome 90.0.4430.212 windows #0-0]          │ consent            │ check          │
[chrome 90.0.4430.212 windows #0-0]          │ termsAndConditions │ check          │
[chrome 90.0.4430.212 windows #0-0]        ✓ Then I can see "Please correct the errors below." error message on the top of the page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I can see registration form error messages
[chrome 90.0.4430.212 windows #0-0]          │ email │ Please enter a valid email │
[chrome 90.0.4430.212 windows #0-0]
[chrome 90.0.4430.212 windows #0-0]     Register new user with various invalid emails (ecx@test)
[chrome 90.0.4430.212 windows #0-0]        ✓ Given I visit home page
[chrome 90.0.4430.212 windows #0-0]        ✓ When I navigate to registration page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I fill out registration form with:
[chrome 90.0.4430.212 windows #0-0]          │ title              │ miss     │
[chrome 90.0.4430.212 windows #0-0]          │ firstName          │ Jane     │
[chrome 90.0.4430.212 windows #0-0]          │ lastName           │ Doe      │
[chrome 90.0.4430.212 windows #0-0]          │ email              │ ecx@test │
[chrome 90.0.4430.212 windows #0-0]          │ password           │ 123456   │
[chrome 90.0.4430.212 windows #0-0]          │ confirmPassword    │ 123456   │
[chrome 90.0.4430.212 windows #0-0]          │ consent            │ check    │
[chrome 90.0.4430.212 windows #0-0]          │ termsAndConditions │ check    │
[chrome 90.0.4430.212 windows #0-0]        ✓ Then I can see "Please correct the errors below." error message on the top of the page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I can see registration form error messages
[chrome 90.0.4430.212 windows #0-0]          │ email │ Please enter a valid email │
[chrome 90.0.4430.212 windows #0-0]
[chrome 90.0.4430.212 windows #0-0]     Register new user with various invalid emails (ecx@)
[chrome 90.0.4430.212 windows #0-0]        ✓ Given I visit home page
[chrome 90.0.4430.212 windows #0-0]        ✓ When I navigate to registration page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I fill out registration form with:
[chrome 90.0.4430.212 windows #0-0]          │ title              │ miss   │
[chrome 90.0.4430.212 windows #0-0]          │ firstName          │ Jane   │
[chrome 90.0.4430.212 windows #0-0]          │ lastName           │ Doe    │
[chrome 90.0.4430.212 windows #0-0]          │ email              │ ecx@   │
[chrome 90.0.4430.212 windows #0-0]          │ password           │ 123456 │
[chrome 90.0.4430.212 windows #0-0]          │ confirmPassword    │ 123456 │
[chrome 90.0.4430.212 windows #0-0]          │ consent            │ check  │
[chrome 90.0.4430.212 windows #0-0]          │ termsAndConditions │ check  │
[chrome 90.0.4430.212 windows #0-0]        ✓ Then I can see "Please correct the errors below." error message on the top of the page
[chrome 90.0.4430.212 windows #0-0]        ✓ And I can see registration form error messages
[chrome 90.0.4430.212 windows #0-0]          │ email │ Please enter a valid email │
[chrome 90.0.4430.212 windows #0-0]
```
But I see this:  

![allure](https://user-images.githubusercontent.com/6464981/119222427-a05a6f80-baf4-11eb-8920-9c9f368d6248.png)

**Comment by BorisOsipov** [comment]
@mareru you're confusing `Data table`(way to pass args to step definition) and `Examples`(way to pass args to scenario outline). It is different things in cucumber.

>But I see this:
You see exactly what you have to see. You see one of scenarios. Other should be in the same suite as independent scenarios.
Like this
![image](https://user-images.githubusercontent.com/6514276/119222777-b1a87980-bafe-11eb-821e-b4e6ab68ba00.png)

**Comment by BorisOsipov** [comment]
@mareru spec report and allure reporter are completely different things. Allure can not behave as spec reporter in most cases.

**Comment by mareru** [comment]
@BorisOsipov 
I am not sure that I follow you.
I am aware that for each outline I will have different record in the allure report. That is what I already have and expect:

![allure2](https://user-images.githubusercontent.com/6464981/119223102-dfd68b00-baf7-11eb-8b43-614a51bd2507.png)

What I also expect but don't have is the data from Data Table in this case (the thing I already reported).  
I just wanted to outline that those 2 (Data Tables and Scenario Outlines) can be combined and in that case I would also expect to see data from Data Table but in place of `<invalidEmail>` variable amongst other data from the Data Table I would like to see in the allure report actual value (e.g `ecx 1@test.com, ecx@, ecx@test`).  
I wanted to emphasize this because I thought it might influence the future implementation of support to the Cucumber Data Tables.

**Comment by BorisOsipov** [comment]
@mareru get it. I think it will be and should not really influence implementation.

**Comment by mareru** [comment]
@BorisOsipov sorry, I didn't express myself in a clear way. :)

**Comment by praveendvd** [comment]
@BorisOsipov  ,  @christian-bromann , @mareru 

![image](https://user-images.githubusercontent.com/45095911/123811726-69455e00-d8eb-11eb-921a-cf2237780cff.png)


Steps:

install csv module

`npm install csv`

now in hook add : 

```

    afterStep: async function (test, scenario, { error, duration, passed }) {
        //throw (["Error isError isError isError is ",JSON.stringify(test),JSON.stringify(scenario)])
        if (error) {
            await browser.takeScreenshot();
        }
        const stringify = require('csv-stringify')
        let data = test["argument"]? test["argument"]["dataTable"]["rows"].map((a) => a.cells.map((a) => a.value)) : undefined
        console.log(data)
        data ?  stringify(data ,function(err, records){
             AllureReporter.addAttachment("Data Table", records, "text/csv")
          }) : null
        
    }
    
```    


Question to @christian-bromann  

should i add a method to Allure report  like `AllureReporter.addDataTable(test)` and make all this logic inside that method instead of forcing user to do it ?

**Comment by BorisOsipov** [comment]
>should i add a method to Allure report like AllureReporter.addDataTable(test)

nope. I should be done automatically in reporter `onTestStart` or similar.

**Comment by praveendvd** [comment]
> > should i add a method to Allure report like AllureReporter.addDataTable(test)
> 
> nope. I should be done automatically in reporter `onTestStart` or similar.

https://github.com/webdriverio/webdriverio/pull/7081 @christian-bromann  and @BorisOsipov  i have added the logic to onteststart, and created a pull request

**Comment by christian-bromann** [comment]
Fixed in #7081, thanks @praveendvd
